Makefile

```diff
@@ -12,6 +12,19 @@ aw-server:
 aw-webui:
 	make -C ./aw-webui build
 
+install-git-hooks:
+	@printf "make pre-commit\n" > .git/hooks/pre-commit
+	@printf "make pre-push\n" > .git/hooks/pre-push
+	@chmod +x .git/hooks/pre-commit .git/hooks/pre-push
+	@printf "Hooks installed\n"
+
+pre-commit:
+	@cargo fmt -- --check || (printf "Error: Run cargo fmt before committing\n"; exit 1)
+
+pre-push: pre-commit
+	@cargo clippy || (printf "Error: Clippy reported error(s)\n"; exit 1)
```

Comment:
I think it would make more sense if this simply printed the error, the custom error message doesn't say anything useful but clippy itself does so might as well just let it print.

Reply:
Currently the project has so many clippy warnings that it's might be unclear that an error has cancelled the commit instead of just having the warnings printed. Maybe the hook here should only test for errors anyways?

Reply:
It's pre-push (not post) so it should still be clear that this is an warning rather than error.